src/lib/tank.js

```diff
@@ -54,67 +54,61 @@ const ensureFundsFor = async (
   askForFunding,
   gotFunding
 ) => {
-  let balance = toBN(await web3.eth.getBalance(address));
+  const balance = toBN(await web3.eth.getBalance(address));
   cost = toBN(cost);
 
-  if (cost.gt(balance)) {
-    try {
-      if (point !== null) {
-        const fundsRemaining = await remainingTransactions(point);
-        if (fundsRemaining < signedTxs.length) {
-          throw new Error('tank: request invalid');
-        }
-      } else {
-        console.log('tank: skipping remaining-funds check');
-        //TODO if we can't always (easily) provide a point, and the
-        //     fundTransactions call is gonna fail anyway, should we maybe
-        //     just not bother doing this check in the first place?
-      }
+  if (balance.gte(cost)) {
+    console.log(
+      `tank: already have sufficient funds: ${address} proposed ` +
+        `a transacton(s) that will cost ${cost.toString()}wei but it ` +
+        `already has ${balance.toString}wei`
+    );
+    return false;
+  }
 
-      const res = await fundTransactions(signedTxs);
-      if (!res.success) {
-        throw new Error('tank: request rejected', res);
-      } else {
-        await waitForTransactionConfirm(web3, res.txHash);
-        let newBalance = await web3.eth.getBalance(address);
-        console.log(
-          'tank: funds have confirmed',
-          balance.toString(),
-          newBalance
-        );
-        return true;
+  try {
+    // TODO: if we can't always (easily) provide a point, and the
+    // fundTransactions call is gonna fail anyway, should we maybe
+    // just not bother doing this check in the first place?
+    if (point !== null) {
+      const fundsRemaining = await remainingTransactions(point);
+      if (fundsRemaining < signedTxs.length) {
+        throw new Error('tank: request invalid');
       }
-    } catch (e) {
-      console.log('tank: funding failed', e);
-      await waitForBalance(
-        web3,
-        address,
-        cost.toString(),
-        askForFunding,
-        gotFunding
-      );
+    } else {
+      console.log('tank: no point provided. skipping remaining-funds check...');
     }
-  } else {
+
+    const res = await fundTransactions(signedTxs);
+    if (!res.success) {
+      throw new Error('tank: request rejected', res);
+    }
+
+    await waitForTransactionConfirm(web3, res.txHash);
+
+    const newBalance = await web3.eth.getBalance(address);
     console.log(
-      'tank: already have sufficient funds',
-      cost.toString(),
-      address,
-      balance.toString()
+      `tank: funds have confirmed: ${address} now has ` +
```

Comment:
Kinda tempted to suggest we should do a `newBalance > cost` check here, just in case of weirdness with the funding tx... Sane y/n?

Reply:
seems like we ran into that in bug reports, so probably worth it, yeah

Reply:
I think just fall back to asking for funds in that case? Retrying the gas tank is another option, but not sure if that's objectively better.